src/adapters/driven/database/order/postgres/product-repository.ts

```diff
@@ -43,6 +44,31 @@ export class ProductRepository implements IProductRepository {
     });
   }
 
+  async findByCategory(
+    category: ProductCategory
+  ): Promise<Product[] | null> {
+    const { records } = await this.connection.query({
+      sql: FIND_PRODUCT_BY_CATEGORY,
+      parameters: {
+        category
+      }
+    });
+
+    if (!records.length) {
```

Comment:
```suggestion
    if (!records.length) {
      logger.debug({
        message: 'Product does not exists',
        data: {
          category,
          records
        }
      });

      return [];
    }
```

Aqui pode retornar array vazio quando não encontrar produtos